src/dachshund/candidate.rs

```diff
@@ -365,25 +365,24 @@ impl<'a, TGraph: GraphBase> Candidate<'a, TGraph> {
         visited_candidates: &mut HashSet<u64>,
     ) -> CLQResult<Vec<Self>> {
         assert!(!visited_candidates.contains(&self.checksum.unwrap()));
-        let mut tie_counts : Vec<(NodeId, usize)> = self.neighborhood
+        let tie_counts : Vec<(NodeId, usize)> = self.neighborhood
             .iter().map(|(&node_id, &edge_count)| (node_id, edge_count)).collect();
 
-        // sort by number of ties, with node_id as tie breaker for deterministic behaviour
-        // [TODO] Instead of sorting the entire list, use a min heap to keep the
-        // top nodes_to_search options.
-        tie_counts.sort_by_key(|k| (Reverse(k.1), k.0));
+        let mut h = BinaryHeap::new();
```

Comment:
would it be better to use `BinaryHeap::with_capacity(num_to_search+1)`?

Reply:
Yes. This is definitely the right approach. Practically speaking, a typical size is something like 20, so we don't save much, but we absolutely should allocate with the known capacity up front.